src/vs/workbench/contrib/terminal/test/browser/xterm/shellIntegrationAddon.test.ts

```diff
@@ -211,6 +211,57 @@ suite('ShellIntegrationAddon', () => {
 			mock.verify();
 		});
 	});
+	suite('BufferMarkCapability', async () => {
+		test('SetMark', async () => {
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, 'foo');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, '\x1b]633;SetMark;\x07');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), true);
+		});
+		test('SetMark - ID', async () => {
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, 'foo');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, '\x1b]633;SetMark;1;\x07');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), true);
+		});
+		test('SetMark - hidden', async () => {
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, 'foo');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, '\x1b]633;SetMark;;true\x07');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), true);
+		});
+		test('SetMark - hidden & ID', async () => {
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, 'foo');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, '\x1b]633;SetMark;1;true\x07');
```

Comment:
What I expected the valid sequence to look like `\x1b]633;SetMark;Id=1;Hidden\x07`